shotover-proxy/src/transforms/cassandra/sink_cluster/topology.rs

```diff
@@ -183,6 +227,174 @@ async fn fetch_current_nodes(
+            })),
+            tx
+        )?;
+
+        let response = rx.await?.response?;
+        into_keyspaces(response, data_center)
+    }
+
+    fn into_keyspaces(
+        mut response: Message,
+        data_center: &str,
+    ) -> Result<HashMap<String, KeyspaceMetadata>> {
+        if let Some(Frame::Cassandra(frame)) = response.frame() {
+            match &mut frame.operation {
+                CassandraOperation::Result(CassandraResult::Rows { rows, .. }) => rows
+                    .iter_mut()
+                    .map(|row| build_keyspace(row, data_center))
+                    .try_collect(),
+                operation => Err(anyhow!(
+                    "keyspace query returned unexpected cassandra operation: {:?}",
+                    operation
+                )),
+            }
+        } else {
+            Err(anyhow!("Failed to parse keyspace query response"))
+        }
+    }
+
+    fn build_keyspace(
+        row: &mut Vec<MessageValue>,
+        data_center: &str,
+    ) -> Result<(String, KeyspaceMetadata)> {
+        let metadata = if let Some(MessageValue::Varchar(string)) = row.pop() {
+            let replication: JsonValue = serde_json::from_str(&string).map_err(|error| {
+                anyhow!(format!(
+                    "Error parsing replication. Error: {} Replication: {}",
+                    error, string
+                ))
+            })?;
+
+            let replication_strategy = match replication {
+                JsonValue::Object(properties) => build_replication_strategy(properties)?,
+                _ => {
+                    return Err(anyhow!(format!(
+                        "Error parsing replication strategy: {}",
+                        replication
+                    )))
+                }
+            };
+
+            let replication_factor = match replication_strategy {
+                ReplicationStrategy::SimpleStrategy { replication_factor } => replication_factor,
+                ReplicationStrategy::NetworkTopologyStrategy {
+                    datacenter_replication_factor,
+                } => *datacenter_replication_factor.get(data_center).unwrap_or(&0),
+                _ => 0,
+            };
+
+            KeyspaceMetadata { replication_factor }
+        } else {
+            return Err(anyhow!(
+                "system_schema.keyspaces.replication is not a varchar"
+            ));
+        };
+
+        let name = if let Some(MessageValue::Varchar(name)) = row.pop() {
+            name
+        } else {
+            return Err(anyhow!("system_schema_keyspaces.name"));
+        };
+
+        Ok((name, metadata))
+    }
+
+    fn build_replication_strategy(
+        mut properties: Map<String, JsonValue>,
+    ) -> Result<ReplicationStrategy> {
+        match properties.remove("class") {
+            Some(JsonValue::String(class)) => Ok(match class.as_str() {
+                "org.apache.cassandra.locator.SimpleStrategy" | "SimpleStrategy" => {
```

Comment:
Why are there 2 possible values for this?

Reply:
I checked driver implementations and noticed they did this, I'd imagine there is some cases where we don't get the fully qualified name.

Reply:
I tracked the source down to this https://github.com/scylladb/scylla-rust-driver/pull/327 but I cant see an appropriate justification for the change.
I also experimented locally with both cassandra and scylla, and I cant find a way to make `system_schema.keyspaces` output in that shortened form.

Unless you have any ideas as to why its there, shall we remove it?

Reply:
Actually, it seems like it gets output in that format when `DESCRIBE KEYSPACE` is used.
Not sure if thats actually possible anymore, it seems to be deprecated and/or DSE specific but its probably worth keeping the shorter form around just in case the code gets repurposed for that use in the future.